wrapper/src/c_sharp_wrapper.cpp

```diff
@@ -386,13 +390,24 @@ bool CSharpWrapper::saveAssetBundles(
 
 		if (inputStream.is_open())
 		{
-			std::ostringstream ss;
-			ss << inputStream.rdbuf();
-			auto s = ss.str();
 			boost::filesystem::path path(assetFiles[i]);
 			std::string fileName = path.filename().string();
 			fileName = "/" + databaseName + "/" + projectName + "/" + fileName;
-			webBuffers.geoFiles[fileName] = std::vector<uint8_t>(s.begin(), s.end());
+
+			std::ostringstream ss; // memory stream containing the binary data to write to the database
+
+			// All unity asset bundles are gzipped by default
+
+			boost::iostreams::filtering_streambuf<boost::iostreams::input> in;
+			in.push(boost::iostreams::gzip_compressor());
+			in.push(inputStream);
+			boost::iostreams::copy(in, ss);
+
+			repo::core::model::RepoBSONBuilder builder;
```

Comment:
@sebjf 

I wouldn't do it this way. This is quite difficult to read and making the way we store metadata data about a file highly coupled with the functions/a classes that are simply passing in the files.

I think this could be a lot simpler (Unless i'm missing something, which I'm often am!). I think the file manager can manage the gzipping and it should just be a flag on the function https://github.com/3drepo/3drepobouncer/blob/master/bouncer/src/repo/core/handler/fileservice/repo_file_manager.h#L60

If this is set to true, it would know to append the gzip flag onto the metadata of the file ref.

We could just gzip all the geo files we export, or it could be a flag in the exporter to indicate whether we want to gzip this. (Persumably we benefit from gzipping the SRCs too, but I'm not sure how unreal handles those requests!)

And it would also make gzipping the JSON caches much easier to do if we want to implement it in the future.

Reply:
Hi @carmenfan, this should be resolved now!